### PR TITLE
feat(org): Add resource `cloudavenue_org_role`

### PR DESCRIPTION
**Pull request by dmicheneau**
If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### How has this code been tested

```
TF_ACC=1 go test -v ./internal/tests -run TestAccOrgRoleResource
=== RUN   TestAccOrgRoleResource
--- PASS: TestAccOrgRoleResource (8.96s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests     8.976s
```

### CODE REVIEW

File: docs/resources/org_role.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `description` (String) Optional description of the role
+- `name` (String) A name for the Role
+- `rights` (Set of String) Optional rights of the role
```
Comment (azrod) [inline docs/resources/org_role.md:37]: This is `optional` or `required` ?

File: internal/provider/org/role_resource.go [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &orgRoleResource{}
+	_ resource.ResourceWithConfigure   = &orgRoleResource{}
+	_ resource.ResourceWithImportState = &orgRoleResource{}
+)
+
+// NewRoleResource is a helper function to simplify the provider implementation.
```
Comment (azrod) [inline internal/provider/org/role_resource.go:28]:
```suggestion
// NewOrgRoleResource is a helper function to simplify the provider implementation.
```

File: internal/provider/org/role_resource.go [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+func NewOrgRolesResource() resource.Resource {
```
Comment (azrod) [inline internal/provider/org/role_resource.go:29]:
```suggestion
func NewOrgRoleResource() resource.Resource {
```

File: internal/provider/org/role_resource.go [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+		resp.Diagnostics.AddError("[role create] Error retrieving Org", err.Error())
+		return
+	}
+
+	// Check rights are valid
+	rights := make([]govcdtypes.OpenApiReference, 0)
+	for _, right := range plan.Rights.Elements() {
+		rg := right.String()
+		rg = strings.Trim(rg, "\"")
+		x, err := adminOrg.GetRightByName(rg)
```
Comment (azrod) [inline internal/provider/org/role_resource.go:137]:
```suggestion
		x, err := adminOrg.GetRightByName(strings.Trim(right.String(), "\""))
```

File: internal/provider/org/role_resource.go [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+
+	// Get the role
+	if state.ID.IsNull() {
+		role, err = adminOrg.GetRoleByName(state.Name.ValueString())
+	} else {
+		role, err = adminOrg.GetRoleById(state.ID.ValueString())
+	}
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			log.Printf("[DEBUG] Unable to find role. Removing from tfstate")
```
Comment (azrod) [inline internal/provider/org/role_resource.go:228]: Please use `tflog.Info` instead of `log.Printf`

File: internal/provider/org/role_resource.go [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
```
Comment (azrod) [inline internal/provider/org/role_resource.go:299]: Please use `tflog.Info` instead of `log.Printf`

File: internal/provider/org/role_resource.go [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("[role update] Error retrieving role", err.Error())
+		return
+	}
+
+	// Update the role Name or Description
+	if (plan.Name != state.Name) || (plan.Description != state.Description) {
```
Comment (azrod) [inline internal/provider/org/role_resource.go:347]:
```suggestion
	if !plan.Name.Equal(state.Name) || !plan.Description.Equal(state.Description) {
```

File: internal/provider/org/role_resource.go [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+			return
+		}
+	}
+
+	// Check rights are valid
+	rights := make([]govcdtypes.OpenApiReference, 0)
+	for _, right := range plan.Rights.Elements() {
+		rg := right.String()
+		rg = strings.Trim(rg, "\"")
+		x, err := adminOrg.GetRightByName(rg)
```
Comment (azrod) [inline internal/provider/org/role_resource.go:362]:
```suggestion
		x, err := adminOrg.GetRightByName(strings.Trim(right.String(), "\""))
```

File: internal/provider/org/role_resource.go [hunk 9]
```diff
@@ -0,0 +1,428 @@
...
+				resp.Diagnostics.AddError("[role update] Error removing role rights", err.Error())
+				return
+			}
+		}
+	}
+
+	// Set Plan state
+	plan = &orgRoleResourceModel{
+		ID:          types.StringValue(role.Role.ID),
+		Name:        plan.Name,
```
Comment (azrod) [inline internal/provider/org/role_resource.go:410]:
```suggestion
		Name:       types.StringValue(role.Role.Name),
```

File: internal/provider/org/role_resource.go [hunk 10]
```diff
@@ -0,0 +1,428 @@
...
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+//go:generate go run github.com/FrangipaneTeam/tf-doc-extractor@latest -filename $GOFILE -example-dir ../../../examples -resource
+func (r *orgRoleResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Import from Api
```
Comment (azrod) [inline internal/provider/org/role_resource.go:426]:
```suggestion
```

**Review by azrod — CHANGES_REQUESTED** [review]

File: internal/provider/org/role_resource.go (hunk 4 above)
Comment (dmicheneau) [reply internal/provider/org/role_resource.go]: Need rg after, and i suggest for keep visibility:
rg = strings.Trim(right.String(), "\"")
x, err := adminOrg.GetRightByName(rg)

File: internal/provider/org/role_resource.go (hunk 8 above)
Comment (dmicheneau) [reply internal/provider/org/role_resource.go]: same thing above

File: internal/provider/org/role_resource.go [hunk 11]
```diff
@@ -0,0 +1,425 @@
...
+	if err != nil {
+		resp.Diagnostics.AddError("[role create] Error retrieving Org", err.Error())
+		return
+	}
+
+	// Check rights are valid
+	rights := make([]govcdtypes.OpenApiReference, 0)
+	for _, right := range plan.Rights.Elements() {
+		rg := strings.Trim(right.String(), "\"")
+		x, err := adminOrg.GetRightByName(rg)
```
Comment (azrod) [inline internal/provider/org/role_resource.go:136]: `rg` assignement is unnecessary 

```suggestion
		x, err := adminOrg.GetRightByName(strings.Trim(right.String(), "\"")
)
```

File: internal/provider/org/role_resource.go [hunk 12]
```diff
@@ -0,0 +1,425 @@
...
+			resp.Diagnostics.AddError("[role update] Error updating role", err.Error())
+			return
+		}
+	}
+
+	// Check rights are valid
+	rights := make([]govcdtypes.OpenApiReference, 0)
+	for _, right := range plan.Rights.Elements() {
+		rg := strings.Trim(right.String(), "\"")
+		x, err := adminOrg.GetRightByName(rg)
```
Comment (azrod) [inline internal/provider/org/role_resource.go:360]: `rg` assignement is unnecessary 

```suggestion
	x, err := adminOrg.GetRightByName(strings.Trim(right.String(), "\""))
```

**Review by azrod — CHANGES_REQUESTED** [review]

**Review by azrod — APPROVED** [review]

**Pull request merged**